src/router.js

```diff
@@ -69,19 +69,11 @@ const initialLocale = () => {
 };
 
 // Implements the restoreSession meta field.
-const restoreSession = async (to) => {
```

Comment:
Now that the logic here is slightly more complicated, I've moved most of this to `util/session.js`. I also think it's good to have most session-related logic consolidated in that file.